intro_to_ada.rst

```diff
@@ -3479,32 +3479,802 @@ Predefined exceptions
+entries will be accepted by using the keyword :ada:`accept`. A task will
+process its statements until it reaches an :ada:`accept` statement. At
+this point, it will wait for the main task or application to synchronize
+with the task. In other words:
+
+- the task will :ada:`accept` a synchronization to happen and it will be
+  waiting in that :ada:`entry` point.
+
+- the main application will call the :ada:`entry` point in order to
+  synchronize with the task.
+
+This synchronization between an application and a task via an :ada:`entry`
+point is called rendez-vous. In the application that synchronizes with the
+task, the rendez-vous is triggered by a call to the :ada:`entry` point ---
+similar to a procedure call.
+
+.. code-block:: ada
+
+    --% run_file: Show_Rendezvous.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Rendezvous is
+
+       task T is
+          entry Start;
+       end T;
+
+       task body T is
+       begin
+          accept Start; -- Waiting for somebody to call the entry
+          Put_Line ("In T");
+       end T;
+
+    begin
+       Put_Line ("In Main");
+       T.Start; --  Calling T's entry
+    end Show_Rendezvous;
+
+In this example, we declare an entry point ``Start`` for the task ``T``.
+In the task body, we use this entry point by using :ada:`accept Start`.
+When the task reaches this point, it will wait for the main application to
+synchronize. This synchronization happens in the ``T.Start`` statement.
+After the synchronization is finished, the main application and task ``T``
+will run concurrently until they synchronize again when the main
+application reaches its end.
+
+Note that the entry point may be used to perform more than just a simple
+task synchronization: we may also perform multiple statements during
+the time both tasks are synchronized. This is achieved by using a
+:ada:`do ... end` block. For the previous example, we would simply write
+:ada:`accept Start do <statements>; end;`. We will use this kind of block
+in the next example.
+
+There is no limit for the amount of times an entry point might be
+accepted. In fact, we could create a loop and accept the same entry point
+over and over. For example:
+
+.. code-block:: ada
+
+    --% run_file: Show_Rendezvous_Loop.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Rendezvous_Loop is
+
+       task T is
+          entry Start;
+       end T;
+
+       task body T is
+          Cnt : Integer := 0;
+       begin
+          loop
+             accept Start do
+                Cnt := Cnt + 1;
+             end Start;
+             Put_Line ("In T's loop (" & Integer'Image (Cnt) & ")");
+          end loop;
+       end T;
+
+    begin
+       Put_Line ("In Main");
+
+       for I in 1 .. 4 loop
+          T.Start; --  Calling T's entry multiple times
+       end loop;
+
+       abort T; --  Aborting T in order to exit loop
```

Comment:
That's not the proper way to do it when you have an "infinite loop" task. You should use `select accept ... or terminate` https://en.wikibooks.org/wiki/Ada_Programming/Tasking#Termination

That follows the general design guideline in threaded applications that cooperative termination is better than prescriptive termination.

Reply:
Yes, I am aware that ```abort``` is awful in this context. I was actually reluctant to use it. Also, I thought about using ```select accept``` as you're saying. However, this was not included in any of your slides, so I thought that you didn't want to use it in the _intro_ course. If we decide to use,I think we should have a section that explains the concept. Do you agree?